Orange/data/variable.py

```diff
@@ -17,15 +22,173 @@ def make_variable(cls, compute_value, *args):
     return cls.make(*args)
 
 
-class VariableMeta(type):
-    # noinspection PyMethodParameters
-    def __new__(mcs, name, *args):
-        cls = type.__new__(mcs, name, *args)
-        if not hasattr(cls, '_all_vars') or cls._all_vars is Variable._all_vars:
-            cls._all_vars = {}
-        if name != "Variable":
-            Variable._variable_types.append(cls)
-        return cls
+def is_discrete_values(values):
```

Comment:
What makes a certain attribute discrete? The current implementation says it is discrete when:
* it has no more than `max_values = max(sqrt(n_instances), 26)` distinct values (26 is arbitrary; adapted to fit one particular test case);
* if numeric, all values are integers (specified without a floating point, i.e. `4`, not `4.0`) between 0 and `max_values`.

I'd wish to simplify this to:
* has no more than `max_values`;
* if numeric, all of whole numbers (e.g. `4.0`, `42` ...);

but in that case, some more test cases would need to be adapted.

@janezd, @astaric, @lanzagar: comments?